Makefile

```diff
@@ -63,18 +63,14 @@ install: build ## Build and install the binary with the current source code. Use
 ##@ Development
 
 .PHONY: generate
-generate: generate-testdata generate-docs ## Update/generate all mock data. You should run this commands to update the mock data after your changes.
+generate: generate-testdata check-docs ## Update/generate all mock data. You should run this commands to update the mock data after your changes.
```

Comment:
The target generate has the purpose to generate only and not check
The check is used in the CI only for we ensure that the docs were properly generate.

Reply:
Sounds good!
I just follow your idea and make the corresponding updates:
- Create a new `hack/docs/generate.sh` that builds the `kb executable` from source code and run the go code to re-generate docs samples
- Modify the target `generate-docs` to call `hack/docs/generate.sh`
- Modify the target `check-docs` to drop unnecessary import and call `hack/docs/generate.sh`

Let me know how you'd like this or I'm understanding wrong :)